Abstract
Despite release of the GRCh38 human reference genome over seven years ago, GRCh37 remains more widely used by most research and clinical laboratories. To date, no study has quantified the impact of utilizing different reference assemblies for the identification of variants associated with rare and common diseases from large-scale exome sequencing data. By calling variants on both the GRCh37 and GRCh38 references, we identified single-nucleotide variants (SNVs) and insertion-deletions (INDELs) in 1,572 exomes from participants with Mendelian diseases and their family members. We found that a total of 1.5% of SNVs and 2.0% of INDELs were discordant when different references were used. Notably, 76.6% of the discordant variants were clustered within discrete DISCordant REference Patches (DISCREPs) comprising only 0.9% of loci targeted by exome sequencing. These DISCREPs were enriched for genomic elements including segmental duplications, fix patch sequences, and loci known to contain alternate haplotypes. We identified 206 genes significantly enriched for discordant variants, most of which were in DISCREPs and caused by multi-mapped reads on the reference assembly that lacked the variant call. Among these 206 genes, eight are implicated in known Mendelian diseases and 53 are associated with common phenotypes from genome-wide association studies. In addition, variant interpretations could also be influenced by the reference after lifting-over variant loci to another assembly. Overall, we identified genes and genomic loci affected by reference assembly choice, including genes associated with Mendelian disorders and complex human diseases that require careful evaluation in both research and clinical applications. 

Alignment, variant calling, and filtering
We generated variant calls for each of the 1,572 samples using both the GRCh37 and GRCh38 human reference assemblies. Specifically, for the GRCh37 variant calls, we selected the hs37d5 reference genome that includes the revised Cambridge reference sequence of the human mitochondrial DNA, human herpesvirus 4 type 1 sequences, and the concatenated decoy sequences;30 and for GRCh38 we selected the full analysis set (GenBank assembly accession: GCA_000001405.15) with decoy sequences, alternate loci scaffolds, and HLA sequences. Both of these references have been used by phase 3 of the 1000 Genomes Project.30 Through a routine, functional equivalent genome sequencing analysis pipeline implemented at the Human Genome Sequencing Center,31 alignment was performed using BWA-MEM32 against the two reference assemblies for each sample followed by Insertion / deletion (INDEL) Realignment and Base Quality Score Recalibration using GATK.9 Variant calling was then performed using xAtlas10 for single-nucleotide variants (SNVs) and INDELs independently to generate gvcf files, which then underwent joint-calling via GLnexus33 (default parameters; autosomal variants only). Notably, we used the same parameters to generate variant call sets on GRCh37 and GRCh38.
We then performed a series of quality controls (QC) following previously published procedures and criteria34,35 to ensure the quality of the aggregated variant calls for SNVs and INDELs. First, for each variant in each sample, we assigned a missing genotype (./.) to the variant if the variant fulfilled one of the following criteria: read depth less than 15; allelic ratio less than 0.25 or greater than 0.75; or non-PASS variant from xAtlas variant call results. We then filtered out variants with a missing genotyping rate greater than 15% to ensure variant quality. This aggressive filtering strategy prevents the identification of stochastic differential variant calls between the two assemblies due to poor variant quality. We kept variants within 100bp to the boundaries of the exome capture targets. In addition, we performed sample-wise QC by ensuring that all samples had a missing genotyping rate <10% and a heterozygosity rate <3 standard deviations. We also calculated pairwise identity by descent across all individuals using PLINK36 to ensure the sample relatedness matches the reported pedigree structures. 

Variant lift-over and comparisons
Variant loci were lifted-over from one reference assembly (source) to the other (target) using the LiftOver tool and the corresponding chain files obtained from the UCSC genome browser.37 We first removed monomorphic variants with alternative allele frequency (AF) of one (AF=1) from both call sets. This removal prevents the comparison of AF=1 variants in the source assembly to AF=0 variants in the target assembly (which did not exist in the filtered vcf file) in cases where the reference and the alternative alleles swap between GRCh37 and GRCh38. During lift-over from GRCh37 to GRCh38, and from GRCh38 to GRCh37, a set of “unliftable” variants was identified on GRCh37 and GRCh38, respectively, due to unmatched loci on the target assembly and were considered as one source of discordant variants. Next, in each sample we determined the genotypic concordance of the “liftable” variants between GRCh37 and GRCh38. Specifically, variants called on GRCh38 were lifted-over to the GRCh37 coordinate and were compared to the variants called on the GRCh37 assembly to identify concordant and discordant variant calls in each sample using vcfeval from the RealTimeGenomics toolbox.38 We also lifted the concordant variants from GRCh37 to GRCh38 to ensure the liftover was bi-directional. Altogether, these analyses resulted in variants that were 1) concordant between GRCh37 and GRCh38, 2) variants found only on GRCh37 (unique to GRCh37), and 3) variants found only on GRCh38 (unique to GRCh38).

Identification of DISCordant REference Patches (DISCREPs)
The analyses were performed on GRCh37 and GRCh38 separately. To identify DISCREPs regions within each assembly, we divided the genome into 10kb windows, counted the number of total distinct variants across all samples in each window, and filtered to keep windows with greater than 10 distinct variants for analysis. Then, in each genomic window, we counted distinct variants uniquely detected by either the GRCh37 or GRCh38 assembly and compared them with the number of distinct concordant variants against the baseline level summed across all windows using one-sided Fisher’s exact test, followed by false-discovery rate (FDR; Benjamini-Hochberg procedure) adjustment. We considered q < 0.01 as a statistically-significant threshold. Illustrations of DISCREPs were conducted using R packages KaryoploteR39 and Circlize.40 

Enrichment analyses for genomic features in DISCREPs
The locations of the following genomic features available for both GRCh37 and GRCh38 were downloaded from the UCSC genome browser:37 simple tandem repeats, microsatellite, segmental duplications, interrupted repeats, known assembly problems, loci with fix patches, loci with alternate haplotypes, loci with known genome assembly differences, and gaps in the assembly. Enrichment analyses for each of the above genomic features within the DISCREPs were performed using an R package LOLA41 based on two-sided Fisher’s exact tests (compared to non- DISCREP genomic windows). The test statistics were adjusted by FDR with q < 0.01 considered to be statistically-significant. For DISCREPs regions that overlapped between GRCh37 and GRCh38, we combined the counts of the genomic windows from both GRCh37 and GRCh38 for the Fisher’s exact tests.

